google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -106,15 +106,22 @@ private AppEngineUtil() {
       @NotNull Project project) {
     List<ModuleDeploymentSource> moduleDeploymentSources = Lists.newArrayList();
 
+    boolean hasNonAppEngineStandardJavaModules = false;
+
     for (Module module : ModuleManager.getInstance(project).getModules()) {
-      if (!containsAppEngineStandardArtifacts(project, module)
-          && isJarOrWarMavenBuild(project, module)) {
-        moduleDeploymentSources.add(createMavenBuildDeploymentSource(project, module));
+      if (ModuleType.is(module, JavaModuleType.getModuleType())) {
+        if (!containsAppEngineStandardArtifacts(project, module)) {
+          hasNonAppEngineStandardJavaModules = true;
+          if (isJarOrWarMavenBuild(project, module)) {
+            moduleDeploymentSources.add(createMavenBuildDeploymentSource(project, module));
+          }
+        }
       }
     }
 
-    Artifact[] artifacts = ArtifactManager.getInstance(project).getArtifacts();
-    if (!containsAppEngineStandardArtifacts(project, Arrays.asList(artifacts))) {
+    // I kind of expected the logic for adding the user specific deployment source to be in the
+    // artifact based deployment sources.
+    if (hasNonAppEngineStandardJavaModules) {
```

Comment:
My original thinking here was since the "user specified" source kind of lives outside of the module/project hierarchy (the user can select _anything_) that we'd only show it if the project has no AE standard artifacts. But, on the other hand, this doesn't match the logic I was using for maven based deployment sources.